src/styles/calendar.scss

```diff
@@ -0,0 +1,64 @@
+.calendar {
+  width: 7.1 * $box-width;
+  height: 5.1 * $box-height;
+
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  padding: 10px;
+  margin: 0 auto;
+  padding-top: 50px;
+
+  $days: 0 "mon", 1 "tue", 2 "wed", 3 "thu", 4 "fri", 5 "sat", 6 "sun";
+
+  @each $day-col, $day-name in $days {
+    &--start-day--#{$day-name} {
+      .calendar__box:first-child {
+        margin-left: $day-col * 101px;
+      }
+    }
+  }
+
+  @for $month-day from 28 through $box-col {
+    &--month-length--#{$month-day} {
+      .calendar__box:nth-child(n + #{$month-day + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__box {
+    position: relative;
+    box-sizing: border-box;
+
+    height: $box-height;
+    width: $box-width;
+
+    background-color: #eee;
+    border: 1px solid black;
+    text-align: center;
+
+    transition-duration: 0.5s;
+
+    @for $box from 1 through $box-col {
+      &:nth-child(#{$box})::before {
+        content: "#{$box}";
+        position: absolute;
+        left: 50%;
+        top: 50%;
+        transform: translate(-50%, -50%);
+
+        font-size: 30px;
+        font-family: Arial, Helvetica, sans-serif;
+      }
+    }
```

Comment:
checklist:
Don't use SASS loops for styles that stay the same for all elements of the group, e.g. display or position.